src/zero/module/aou.clj

```diff
@@ -1,23 +1,385 @@
+                                        :release  release
+                                        :uuid     (UUID/randomUUID)
+                                        :version  version
+                                        :wdl      top})
+            table (format "%s_%09d" pipeline id)
+            table_seq (format "%s_id_seq" table)
+            kind  (format (str/join " " ["UPDATE workload"
+                                         "SET pipeline = '%s'::pipeline"
+                                         "WHERE id = %s"]) pipeline id)
+            idx (format "CREATE SEQUENCE %s AS bigint" table_seq)
+            work  (format "CREATE TABLE %s OF %s (PRIMARY KEY (analysis_version_number, chip_well_barcode), id WITH OPTIONS NOT NULL DEFAULT nextval('%s'))" table pipeline table_seq)
+            link-idx-work (format "ALTER SEQUENCE %s OWNED BY %s.id" table_seq table)]
+        (jdbc/update! tx :workload {:items table} ["id = ?" id])
+        ; this is really painful and hideous, but I cannot find a better way to bypass
+        ; https://www.postgresql.org/docs/current/datatype-numeric.html#DATATYPE-SERIAL
+        (jdbc/db-do-commands tx [kind idx work link-idx-work])
+        {:uuid uuid}))))
+
+(comment
+  (jdbc/with-db-transaction [tx (postgres/zero-db-config)]
+                            (let [body {:creator  "rex"
+                                        :cromwell "http://cromwell-gotc-auth.gotc-dev.broadinstitute.org/"
+                                        :project  "gotc-dev"
+                                        :pipeline "AllOfUsArrays"}]
+                              (add-workload! tx body))))
 
 (defn start-workload!
-  "Start the WORKLOAD in the database DB."
-  [tx workload]
-  (->> workload
-       (filter second)
-       (into {})))
+  "Use transaction TX to start the WORKLOAD by UUID. This is simply updating the
+   workload table to mark a workload as 'started' so it becomes append-able."
+  [tx {:keys [uuid] :as body}]
+  (let [now (OffsetDateTime/now)]
+    (jdbc/update! tx :workload {:started now} ["uuid = ?" uuid])))
+
+(defn keep-primary-keys
+  "Only return the primary keys of a SAMPLE."
+  [sample]
+  (select-keys sample [:chip_well_barcode :analysis_version_number]))
+
+(defn remove-existing-samples!
+  "Return set of SAMPLEs that are not registered in workload TABLE using transaction TX."
+  [tx samples table]
+  (letfn [(q [[left right]] (fn [it] (str left it right)))
+          (extract-key-groups [l] [(map :chip_well_barcode l) (map :analysis_version_number l)])
+          (make-query-group [v] (->> v (map (q "''")) (str/join ",") ((q "()"))))
+          (assemble-query [query l] (format query (first l) (last l)))]
+    (let [samples  (map keep-primary-keys samples)
+          existing (->> samples
+                        (extract-key-groups)
+                        (map make-query-group)
+                        (assemble-query (str/join " " ["SELECT * FROM" table "WHERE chip_well_barcode in %s"
```

Comment:
A more secure way to do this query is to construct the parameterized string dynamically (since there is a variable number of samples) and then set each parameter like this! https://github.com/broadinstitute/wfl/blob/ec7cc6bf0f68fa4d81f9499796bc3bbece57f42e/src/zero/api/handlers.clj#L105-L111